pkgs/development/python-modules/cocotb/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub, setuptools, swig, verilog }:
+
+buildPythonPackage rec {
+  pname = "cocotb";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "091q63jcm87xggqgqi44lw2vjxhl1v4yl0mv2c76hgavb29w4w5y";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+  ];
+
+  postPatch = ''
+    patchShebangs bin/*.py
+
+    # POSIX portability (TODO: upstream this)
+    for f in \
+      cocotb/share/makefiles/Makefile.* \
+      cocotb/share/makefiles/simulators/Makefile.*
+    do
+      substituteInPlace $f --replace 'shell which' 'shell command -v'
+    done
+  '';
```

Comment:
another option could be
```
  nativeBuildInputs = [ which ];
```
this will add which to PATH during build

Reply:
If it was just tests, then I should agree, but this would force every external use of cocotb to run simulations using Nix to include `which` as a dependency, which I think is undesirable (they already have to add an appropriate Verilog/VHDL simulator to the PATH).

Reply:
I'm not familiar with cocotb, and the `shell command -v` is probably the proper thing to do.

But which won't become a runtime dependency, just a build-time one.

Reply:
cocotb installs these Makefiles into `$out` and they're used when doing a simulation workflow with it.